src/frontend/src/binder/mod.rs

```diff
@@ -511,6 +495,47 @@ impl Binder {
     }
 }
 
+pub struct PopContextGuard<'a>(&'a mut Binder);
+
+impl<'a> Drop for PopContextGuard<'a> {
+    fn drop(&mut self) {
```

Comment:
since we are implicitly poping context here, maybe add some comments to better illustrate the usage / purpose.